brush-parser/src/parser.rs

```diff
@@ -911,4 +927,21 @@ esac\
 
         Ok(())
     }
+
+    #[test]
+    fn parse_redirection() -> Result<()> {
+        let input = r"echo |& wc";
+
+        let tokens = tokenize_str(input)?;
+        let program = super::token_parser::program(
+            &Tokens {
+                tokens: tokens.as_slice(),
+            },
+            &ParserOptions::default(),
+            &SourceInfo::default(),
+        )?;
+        let snapshot = r#"Program { complete_commands: [CompoundList([CompoundListItem(AndOrList { first: Pipeline { bang: false, seq: [Simple(SimpleCommand { prefix: None, word_or_name: Some(Word { value: "echo" }), suffix: None }), Simple(SimpleCommand { prefix: None, word_or_name: Some(Word { value: "wc" }), suffix: None })] }, additional: [] }, Sequence)])] }"#;
```

Comment:
Thanks for adding a test case here. Having `Debug` implemented for human debugging is helpful, but we should avoid taking a programmatic dependency on its specific textual format.

In some other test cases, we match against the `ast::Program` (or more specific AST nodes). Could you go with that approach here?